tests/Microsoft.Identity.Test.Unit/PublicApiTests/AdfsAcceptanceTests.cs

```diff
@@ -0,0 +1,86 @@
+        {
+            await RunAuthCodeFlowAsync(authority, useWithAdfsAuthority: true).ConfigureAwait(false);
+        }
+
+        [DataTestMethod]
+        [DataRow("https://localhost:3001/adfs")]
+        [DataRow("https://localhost:3001/460afc9d-718d-40c8-8d03-954")]
+        [DataRow("https://localhost:3001/contoso.int.test")]
+        public async Task AdfsAuthorityVariants_WithAuthority_Async(string authority)
+        {
+            await RunAuthCodeFlowAsync(authority, useWithAdfsAuthority: false).ConfigureAwait(false);
+        }
+
+        private static async Task RunAuthCodeFlowAsync(string authority, bool useWithAdfsAuthority)
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                var builder = ConfidentialClientApplicationBuilder
+                    .Create("e68c40a5-a8e5-4250-bbea-5b43ab18cf0d");
+
+                builder = useWithAdfsAuthority ?
+                    builder.WithAdfsAuthority(authority) :
+                    builder.WithAuthority(authority);
+
+                var app = builder.WithAdfsAuthority(authority, false)
+                    .WithRedirectUri("http://localhost")
+                    .WithHttpManager(httpManager)
+                    .WithInstanceDiscovery(false)
+                    .WithClientSecret(TestConstants.ClientSecret)
+                    .Build();
+
+
+                httpManager.AddAdfsWithTenantIdMockHandler();
+
+                var result = await app.AcquireTokenByAuthorizationCode(new[] { "https://arm.asz/.default" }, "authcode")
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                var account = await app.GetAccountAsync(result.Account.HomeAccountId.Identifier).ConfigureAwait(false);
+
+                AssertAdfsResult(result, account);
+
+                var result2 = await app.AcquireTokenSilent(new[] { "https://arm.asz/.default" }, account)
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                var account2 = await app.GetAccountAsync(result.Account.HomeAccountId.Identifier).ConfigureAwait(false);
+                AssertAdfsResult(result2, account2);
+                Assert.AreEqual(TokenSource.Cache, result2.AuthenticationResultMetadata.TokenSource);
+            }
+        }
+
+        private static void AssertAdfsResult(AuthenticationResult result, IAccount account)
+        {
+            Assert.AreEqual("460afc9d-718d-40c8-8d03-9540fa56cc2c", result.TenantId);
+            Assert.AreEqual("460afc9d-718d-40c8-8d03-9540fa56cc2c", result.ClaimsPrincipal.FindFirst("tid").Value);
+
+            Assert.AreEqual("localhost", account.Environment);
+            Assert.AreEqual("admin@demo.asz", account.Username);
+            Assert.AreEqual("FTiFcJ97JrNoywo4SSdQjA", account.HomeAccountId.Identifier);
+            Assert.AreEqual("FTiFcJ97JrNoywo4SSdQjA", account.HomeAccountId.ObjectId);
```

Comment:
I'd put these in TestConstants; otherwise when some mock tokens get refactored, test throughout break.

Reply:
This is actually a pretty complex setup, with a token transformation that adds `tid`. So I'm gonna move all test data in this class instead.